### ISSUE TITLE
[Proposal] Evaluating Cross-Lingual Performance and Implicit Translation in Large Language Models

### ISSUE THREAD

**Issue by ka1ku** [uchicago-nlp-course/winter-2026-students#25]
[Kai_Justin_Proposal.pdf](https://github.com/user-attachments/files/25171521/Kai_Justin_Proposal.pdf)

**Comment by darinkeng** [comment]
Strengths:

- Clearly pointed out how models are currently evaluated and how they are actually used in the real world. This project addresses the field's current limitations regarding fairness and reliability
- Probing whether models implicitly translate when reading non-English input is an interesting hypothesis to test!

Weakness:

- You mention you want to evaluate models as black-box system, but these probing problems usually need to look at the hidden states. I would suggest using `TransformerLens` to https://transformerlensorg.github.io/TransformerLens/. When you apply this to a prompt, you can watch the model change its mind as the signal moves up the layers, which should be very useful in your research!
- There's also a library called Logit Lens, I believe this should also be useful: https://nnsight.net/notebooks/tutorials/probing/logit_lens/

Next steps:

- Try to finalize the models and the tasks you want to perform. Create a table reporting the results of the baseline models(You can use a small subset of data if you want, it could be just a toy experiment), and remember to report the table in your weekly blogs!

**Comment by ka1ku** [comment]
[blog_week8.pdf](https://github.com/user-attachments/files/25476399/blog_week8.pdf)

**Comment by darinkeng** [comment]
Strengths
- Love the logit lens technique! By doing this, you can visually check whether there’s implicit translation happening.
- It’s great that you organized the benchmark data into several categories. 

Weakness:
- For the baseline accuracy, you used modern LLMs like GPT-3.5-turbo and Gemini 2.5 Flash. However, you used a much smaller 560-parameter model for internal probing. It is completely understandable to use a smaller model for white-box probing, but a 560M model's internal processing might look different than a massive state-of-the-art model. It would also be worth trying to run it on a larger and more modern LLM to make it comparable to the baseline models. It would also be great to run the BLOOM model as a baseline, just to make it comparable to the probing result.

Next steps:
- Your planned next Since you noted running out of time for the residual stream extraction in Week 7, make sure to make this top priority for this week. If you need any suggestions on the RSA, please let me know! Keep up the good work!

**Comment by ka1ku** [comment]
[blog_week9.pdf](https://github.com/user-attachments/files/25672817/blog_week9.pdf)

**Comment by darinkeng** [comment]
Strengths:
- Unifying the black-box and white-box testing on Mistral-7B-Instruct-v0.2 was a big plus. This makes the white-box analysis now directly explain the black-box results. Great job!
- Fantastic work on all the probing results!

Weakness:
- Although you already have some promising results on the Mistral model, I think testing other model families, e.g, Qwen ot Llama, can further make your research more robust. By doing this you can test if the U-shaped implicit translation curve is a universal mechanism in LLMs or just Mistral specific nature.
- The total collapse of Basque accuracy (0% on reasoning) makes the layer-wise RSA findings for Basque somewhat meaningless, since this shows the model fundamentally lacks proficiency in Basque. Perhaps you can switch to another model that contains Basque knowledge e.g [Llama-eus-8B]( https://huggingface.co/orai-nlp/Llama-eus-8B), which is finetuned on Basque documents and still has strong performance on English tasks.

Next steps:
- You can try the multilingual llama model if possible, then start drafting you final paper. Keep up the good work!

**Comment by ka1ku** [comment]
[paper.pdf](https://github.com/user-attachments/files/25959591/paper.pdf)
